File: backend/app/schemas/communication.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import datetime
from uuid import UUID
from app.models.enums import AnnouncementPriority, AnnouncementScope, AnnouncementStatus

class AnnouncementBase(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    priority: Optional[AnnouncementPriority] = AnnouncementPriority.NORMAL
    scope: Optional[AnnouncementScope] = AnnouncementScope.GLOBAL
    target_criteria: Optional[List[str]] = None
    status: Optional[AnnouncementStatus] = AnnouncementStatus.DRAFT
    published_at: Optional[datetime] = None
    expires_at: Optional[datetime] = None

class AnnouncementCreate(AnnouncementBase):
    title: str
    content: str

class AnnouncementUpdate(AnnouncementBase):
    pass

class AnnouncementInDBBase(AnnouncementBase):
    id: UUID
    title: str
    content: str
    created_by: UUID
    created_at: datetime
    
    class Config:
        from_attributes = True

class Announcement(AnnouncementInDBBase):
    author_name: Optional[str] = None

File: frontend/src/types/communication.ts
import { UUID } from "./common";

export enum AnnouncementPriority {
    LOW = "THAP", // Frontend only
    NORMAL = "BINH_THUONG",
    HIGH = "QUAN_TRONG",
    URGENT = "KHAN_CAP"
}

export enum AnnouncementScope {
    GLOBAL = "TOAN_HE_THONG",
    CAMPUS = "CO_SO", // New scope
    BUILDING = "TOA_NHA",
    FLOOR = "TANG", // Frontend only
    ROOM = "PHONG" // Frontend only
}

export enum AnnouncementStatus {
    DRAFT = "NHAP",
    PUBLISHED = "DA_DANG",
    ARCHIVED = "LUU_TRU"
}

export interface Announcement {
    id: UUID;
    title: string;
    content: string;
    author_name?: string;
    priority: AnnouncementPriority;
    scope: AnnouncementScope;
    status: AnnouncementStatus;
    published_at?: string;
    expires_at?: string;
    target_criteria?: string[];
    created_at: string;
}

export interface AnnouncementCreate {
    title: string;
    content: string;
    priority?: AnnouncementPriority;
    scope?: AnnouncementScope;
    target_criteria?: string[];
    status?: AnnouncementStatus;
    published_at?: string;
    expires_at?: string;
}

export interface AnnouncementUpdate {
    title?: string;
    content?: string;
    priority?: AnnouncementPriority;
    scope?: AnnouncementScope;
    target_criteria?: string[];
    status?: AnnouncementStatus;
    published_at?: string;
    expires_at?: string;
}
